### ISSUE TITLE
ship a bundle

### ISSUE THREAD

**Issue by inmarelibero** [drgomesp/symfony-docker#18]
I'm sorry if this is a stupid question, because I don't know perfectly how docker and Symfony can integrate, but:

**can't this be a bundle?**

I mean: I won't use this as start for a Symfony project, because I want to start from `composer create-project` or whatever, but I'd surely use this as a bundle.

Once I start a project based on this boilerplate, I cannot update it, to have the latest features. But with a bundle, this can be done. And of course, you could ship also the `Symfony Docker Edition` with this bundle as required.

Some desired specs:
- configurable (eg. I want to use a local mysql for tests, but for the `prod` container I want to connect to the real database server; or I want to choose between apache and nginx)
- cli commands

**Comment by drgomesp** [comment]
If your creating a new project, you can simply use:

`composer create-project drgomesp/symfony-docker`

To solve the _keeping the project updated according to the symfony-docker distribution_, you can add a remote to point to this repository:

`git remote add dist https://github.com/drgomesp/symfony-docker.git`

And whenever you want to update it, you can simply run:

`git pull --rebase dist [VERSION]`

Where `VERSION` can be `master`, `1.0.0` or whatever other version/branch.

Let me know if this works, because I'm really interested in finding the best possible solution for this problem too.

**Comment by inmarelibero** [comment]
thanks for the attention ;)

So, realistically I'll never make a rebase on a working project, which has hundreds of commits. Following this, why use bundles, and not incorporate code to rebase, as you suggest? No, it's not realistic.

Also, if I'm already using another "distribution" (like yours), a rebase is simply not an option. Or, what if I change some folders, not following standards? We could catch some feedbacks, but I bet very few people will find rebase an option.

What do you think?

**Comment by drgomesp** [comment]
You make a strong point, @inmarelibero. Could you give me some more details on how that bundle would be structured? Especially because I think Docker is greater than the scope of a bundle in Symfony.

**Comment by inmarelibero** [comment]
well a bundle can be structured as you want in fact.

looking at what it should contain, it would be (correct me if I'm wrong, because I don't know docker well):
- ./Dockerfile
- docker-compose.json
- ./docker

What it should really do is (I propose):
- provide a cli command `app/console docker:run` (like `app/console server:run`) that, reading `app/config/mydockerconfigfile.yml`, would run docker as configured in `app/config/mydockerconfigfile.yml`

Something that just came up in my mind is:

```
currently my project is based on php5, and I'd like to try it on php7, just to see if it runs.
I could prepare a app/config/docker_php_7.yml and run it with the command app/console docker:run
```

**Comment by drgomesp** [comment]
@inmarelibero 

Well, after having experienced a little bit with the flow of updating the projects created based on symfony-docker, I've decided to ship a bundle with the Docker related stuff. This way, people can either integrate the bundle to their existing apps, or create a project from this, and also have the bundle able to update itself through composer.

Here's the task list that I came up with (add more as you like)
- [x] Create the DockerBundle
- [x] Move all Docker related config to bundle
- [x] Create command to move `docker-compose` file from bundle to root (maybe not needed, since we can have a command `docker:build` and `docker:run` and keep the file on the bundle.

Started the work here https://github.com/SymfonyDocker/DockerBundle
